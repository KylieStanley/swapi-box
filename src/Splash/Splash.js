import React, { Component } from 'react';
import fetchData from '../helper/APICalls.js'
import './Splash.scss';

export default class Splash extends Component {
  constructor() {
    super()
    this.state = {
      film: null
    }
  }

  async componentDidMount() {
    const url = 'https://swapi.co/api/films/'
    const films = await fetchData(url)
    const randomNumber = Math.floor(Math.random() * 7)

    this.setState({
      film: films.results[randomNumber]
    })
  }

render() {
  return (
    this.state.film ? <div className='splash'>
      <div className="fade">
        <h1 className='logo'>SWAPi Box</h1>
        <button className='enter-btn' onClick={ this.props.toggleSplash }>$</button>
        <p className='enter-site'>Click to Enter</p>
      </div>
      <section className="star-wars">
        <div className="crawl">
          <h2 className='film-title'>{ this.state.film.title }</h2>
          <p className='film-body'>{ this.state.film.opening_crawl }</p>
        </div>
      </section>
    </div> : <h1>Loading...</h1>
    )
  }
}