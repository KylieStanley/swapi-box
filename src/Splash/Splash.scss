@font-face {
  font-family: 'Star Wars';
  src: url(../Fonts/starjedi/Starjhol.ttf);
}

@font-face {
  font-family: 'Jedi';
  src: url(../Fonts/starjedi/Starjedi.ttf);
}

.splash {
  background: black;
  height: 100vh;
  width: 100vw;
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  overflow: hidden;
}

.logo {
  color: #ffda00;
  font-size: 96px;
  font-family: 'Star Wars';
}

.fade {
  width: 100vw;
  min-height: 70vh;
  position: relative;
  background: linear-gradient(0deg, transparent, black 25%);
  z-index: 1;
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
}

.enter-btn {
  font-family: 'Jedi';
  font-size: 60px;
  background: none;
  color: #56c5ea;
  border: none;
  height: 80px;
  margin-top: -15px;
  animation: glow .3s linear;
}

.enter-site {
  color: #56c5ea;
  margin-top: 0px;
  animation: glow .3s linear;
}

.star-wars {
  display: flex;
  justify-content: center;
  position: relative;
  perspective: 400px;
  height: 30vh;
}

.crawl {
  color: #ffda00;
  position: relative;
  top: 1000px;
  transform-origin: 50% 100%; 
  animation: crawl 87s linear;
  font-weight: bold;
}

.film-title {
  font-size: 80px;
  text-align: center;
}

.film-body {
  font-size: 50px;
  text-align: justify;
  line-spacing: 6px;
}

@keyframes crawl {
  0% {
    top: 300px;
    transform: rotateX(20deg) translateZ(0)
  }
  100% {
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px)
  }
}

@keyframes glow {
    to {
        text-shadow: 0 0 10px #fff;
    }
}