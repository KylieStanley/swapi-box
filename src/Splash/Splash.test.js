import React from 'react'
import { shallow, mount } from 'enzyme'
import Splash from './Splash'

describe('Splash', () => {
  it('should match the snapshot', () => {
    const mockFilms = [
      {
        'characters': [],
        'created': 2014,
        'director': 'George Lucas',
        'edited': 2015,
        'episode_id': 4,
        'opening_crawl': 'this is the crawl',
        'planets': 'Earth',
        'producer': 'George',
        'release_date': 2015,
        'species': ['humans', 'aliens'],
        'starships': ['falcon', 'star fighter'],
        'title': 'A New Hope',
        'url': 'http://this.com',
        'vehicles': ['this vehicle', 'that vehicle']
      }
    ]
    const mockFunction = jest.fn()
    const wrapper = shallow(<Splash films={ mockFilms } toggleSplash={ mockFunction } />)

    expect(wrapper).toMatchSnapshot()
  })
})