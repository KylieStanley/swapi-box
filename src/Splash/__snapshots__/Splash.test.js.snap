// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Splash
    films={
      Array [
        Object {
          "characters": Array [],
          "created": 2014,
          "director": "George Lucas",
          "edited": 2015,
          "episode_id": 4,
          "opening_crawl": "this is the crawl",
          "planets": "Earth",
          "producer": "George",
          "release_date": 2015,
          "species": Array [
            "humans",
            "aliens",
          ],
          "starships": Array [
            "falcon",
            "star fighter",
          ],
          "title": "A New Hope",
          "url": "http://this.com",
          "vehicles": Array [
            "this vehicle",
            "that vehicle",
          ],
        },
      ]
    }
    toggleSplash={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Loading...",
    },
    "ref": null,
    "rendered": "Loading...",
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Loading...",
      },
      "ref": null,
      "rendered": "Loading...",
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
